frontend/src/views/WheresMyGene/components/HeatMap/utils.ts

```diff
@@ -229,7 +229,10 @@ interface CreateYAxisOptionsProps {
 /**
  * Used to calculate text pixel widths. Should be only created once.
  */
-const CTX = document.createElement("canvas").getContext("2d");
+const CTX =
```

Comment:
nextjs 12 somehow invokes this file on the server side, which doesn't have access to `document`, so we need to check `typeof document` now

Reply:
weird...